src/realm/util/network.cpp

```diff
@@ -1155,6 +1155,8 @@ bool Service::IoReactor::wait_and_advance(clock::time_point timeout, clock::time
                     ret++;
                 }
 
+                std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
hmm, i think maybe this should be on line 1151 like

```
if (m_pollfd_slots.size() > 1) {
    ...
} else if (m_wakeup_pipe.is_signaled()) {
    ...
} else {
    sleep()
}
```
Maybe what's going on here is that you don't have any active sync sessions and so you're stuck busy waiting for a sync session to open or the Service to get something posted to it?